keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/resource/ImageHandler.kt

```diff
@@ -46,22 +47,39 @@ class ImageHandler(
 
   override fun desired(resource: Resource<ImageSpec>): Image =
     with(resource.spec) {
+      val artifact = DeliveryArtifact(artifactName, DEB)
       val latestVersion = artifactRepository
-        .versions(DeliveryArtifact(artifactName, DEB))
-        .first()
-      Image(
-        baseAmiVersion = baseImageCache.getBaseImage(baseOs, baseLabel),
-        // TODO: there must be a canonical way of composing this
-        appVersion = latestVersion,
-        regions = regions
-      )
+        .versions(artifact)
+        .firstOrNull() ?: throw NoKnownArtifactVersions(artifact)
+      val baseImage = baseImageCache.getBaseImage(baseOs, baseLabel)
+      runBlocking {
+        val baseAmi = cloudDriver.namedImages(baseImage, "test")
+          .await()
+          .lastOrNull()
+          ?.let { namedImage ->
+            val tags = namedImage
+              .tagsByImageId
+              .values
+              .first { it?.containsKey("base_ami_version") ?: false }
+            if (tags != null) {
+              tags.getValue("base_ami_version")!!
+            } else {
+              null
+            }
+          } ?: throw BaseAmiNotFound(baseImage)
+        Image(
+          baseAmiVersion = baseAmi,
+          appVersion = latestVersion,
+          regions = regions
+        )
+      }
     }
 
   override fun current(resource: Resource<ImageSpec>): Image? =
     runBlocking {
-      cloudDriver.images("aws", resource.spec.artifactName)
+      cloudDriver.namedImages(resource.spec.artifactName, "test")
```

Comment:
I don't know what I was thinking with that endpoint but it doesn't seem to actually exist.

Question: is checking in the `test` account all we need?

Reply:
all images will be in the test account. what matters is the region - images are shared from test + region to other account + region for us.